x/amm/types/message_exit_pool.go

```diff
@@ -46,5 +46,14 @@ func (msg *MsgExitPool) ValidateBasic() error {
 	if err != nil {
 		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid sender address (%s)", err)
 	}
+
+	if msg.ShareAmountIn.IsNil() {
+		return ErrInvalidShareAmountOut
+	}
+
+	if msg.ShareAmountIn.IsNegative() {
+		return ErrInvalidShareAmountOut
```

Comment:
This one is in amount and error thrown is for out

Reply:
Fixed it